src/CgiRequestHandler.cpp

```diff
@@ -104,13 +260,31 @@ void CgiRequestHandler::checkTimeout()
 
 void CgiRequestHandler::onBodyData(std::span<const char> data)
 {
+	if (bufferingRequestBody_)
+	{
+		// Buffer the request body for deferred CGI launch
+		requestBodyBuffer_.append(data.begin(), data.end());
+		return;
+	}
+	
 	if (!cgiHandler_)
 		return;
 	size_t queued = cgiHandler_->queueWrite(data);
 	if (queued > PIPE_WRITE_HIGH_WATER_MARK)
 		manager_.setReadingBody(false);
 }
 
+void CgiRequestHandler::onBodyDone()
+{
+	if (bufferingRequestBody_)
+	{
+		// Now we have the full body, launch CGI with proper CONTENT_LENGTH
+		bufferingRequestBody_ = false;
+		launchCgiProcess();
+	}
+	// For non-chunked requests, body is streamed directly so nothing to do
+}
```

Comment:
`launchCgiProcess` has `if (bufferingRequestBody_)` blocks, setting the flag to false first breaks it. It might be better to move those blocks here instead.